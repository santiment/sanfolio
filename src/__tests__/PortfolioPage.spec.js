/* eslint-env jasmine */
import React from 'react'
import {
  MemoryRouter
} from 'react-router-dom'
import { configure, shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import {
  PortfolioPage,
  PortfoliosNaviagation
} from '../PorfoliosPage'
configure({ adapter: new Adapter() })

describe('PortfolioPage', () => {
  let wrapper
  const portfolios = {
    selected: 0,
    items: [
      {
        name: 'Awesome Portfolio 1',
        money: 23523,
        items: []
      },
      {
        name: 'Portfolio 2',
        money: 234222,
        items: []
      }
    ]
  }

  beforeEach(() => {
    wrapper = shallow(
      <MemoryRouter>
        <PortfolioPage portfolios={portfolios} />
      </MemoryRouter>)
  })

  it('should redirect to invest page, if portfolios items is empty', () => {
    const emptyPortfolios = {
      selected: 0,
      items: []
    }
    const redirectedWrapper = shallow(
      <MemoryRouter>
        <PortfolioPage portfolios={emptyPortfolios} />
      </MemoryRouter>)
    expect(toJson(redirectedWrapper)).toMatchSnapshot()
  })

  it('should render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
    expect(wrapper.length).toEqual(1)
  })

  describe('PortfoliosNaviagation', () => {
    it('should render correctly', () => {
      const navWrapper = mount(
        <MemoryRouter>
          <PortfoliosNaviagation portfolios={portfolios} />
        </MemoryRouter>)
      expect(toJson(navWrapper)).toMatchSnapshot()
      expect(navWrapper.find('ListHeader').text()).toEqual(portfolios.items[0].name)
    })
  })
})
