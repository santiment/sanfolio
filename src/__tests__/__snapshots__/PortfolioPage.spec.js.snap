// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioPage PortfoliosNaviagation should render correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "26sax4",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "26sax4",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      portfolios={
        Object {
          "items": Array [
            Object {
              "items": Array [],
              "money": 23523,
              "name": "Awesome Portfolio 1",
            },
            Object {
              "items": Array [],
              "money": 234222,
              "name": "Portfolio 2",
            },
          ],
          "selected": 0,
        }
      }
    >
      <List
        horizontal={true}
      >
        <div
          className="ui horizontal list"
          role="list"
        >
          <ListItem
            key="0"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <ListContent
                className="profile-link-active"
              >
                <div
                  className="content profile-link-active"
                >
                  <ListHeader>
                    <div
                      className="header"
                    >
                      <Link
                        replace={false}
                        to="/portfolios/Awesome Portfolio 1"
                      >
                        <a
                          href="/portfolios/Awesome Portfolio 1"
                          onClick={[Function]}
                        >
                          Awesome Portfolio 1
                        </a>
                      </Link>
                    </div>
                  </ListHeader>
                </div>
              </ListContent>
            </div>
          </ListItem>
          <ListItem
            key="1"
          >
            <div
              className="item"
              onClick={[Function]}
              role="listitem"
            >
              <ListContent
                className=""
              >
                <div
                  className="content"
                >
                  <ListHeader>
                    <div
                      className="header"
                    >
                      <Link
                        replace={false}
                        to="/portfolios/Portfolio 2"
                      >
                        <a
                          href="/portfolios/Portfolio 2"
                          onClick={[Function]}
                        >
                          Portfolio 2
                        </a>
                      </Link>
                    </div>
                  </ListHeader>
                </div>
              </ListContent>
            </div>
          </ListItem>
        </div>
      </List>
    </Component>
  </Router>
</MemoryRouter>
`;

exports[`PortfolioPage should redirect to invest page, if portfolios items is empty 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "qbuesv",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "qbuesv",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <PortfolioPage
    portfolios={
      Object {
        "items": Array [],
        "selected": 0,
      }
    }
  />
</Router>
`;

exports[`PortfolioPage should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "76i12h",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "76i12h",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <PortfolioPage
    portfolios={
      Object {
        "items": Array [
          Object {
            "items": Array [],
            "money": 23523,
            "name": "Awesome Portfolio 1",
          },
          Object {
            "items": Array [],
            "money": 234222,
            "name": "Portfolio 2",
          },
        ],
        "selected": 0,
      }
    }
  />
</Router>
`;
